### PR TITLE
QALI-515 Statistics factory

### PR DESCRIPTION
**Pull request by ericjesse**

### CODE REVIEW

File: api-common/src/main/kotlin/io/qalipsis/api/meters/Meter.kt [hunk 1]
```diff
@@ -36,14 +33,19 @@ interface Meter<SELF : Meter<SELF>> {
     fun report(configure: ReportingConfiguration<SELF>.() -> Unit): SELF
 
     /**
-     * Get a set of measurements. Should always return the same number of measurements and
-     * in the same order, regardless of the level of activity or the lack thereof.
-     * @return The set of measurements that represents the instantaneous value of this
-     * meter.
+     * Generates a snapshot representing the instantaneous state of the meter. Generating a snapshots
```
Comment (oceenachi) [inline api-common/src/main/kotlin/io/qalipsis/api/meters/Meter.kt:36]:
```suggestion
     * Generates a snapshot representing the instantaneous state of the meter. Generating a snapshot
```

File: api-common/src/main/kotlin/io/qalipsis/api/meters/MeterType.kt [hunk 2]
```diff
@@ -17,18 +17,20 @@
 package io.qalipsis.api.meters
 
 /**
- * Possibilities of [io.qalipsis.api.meters] that are available.
+ * Possibilities of [io.qalipsis.api.meters.Meter] that are available.
```
Comment (oceenachi) [inline api-common/src/main/kotlin/io/qalipsis/api/meters/MeterType.kt:20]:
```suggestion
 * Possibilities of [io.qalipsis.api.meters.Meter] that is available.
```

File: api-common/src/main/kotlin/io/qalipsis/api/meters/MeterSnapshot.kt [hunk 3]
```diff
@@ -21,16 +21,25 @@ import java.time.Instant
...
+
+    val meterId: Meter.Id
 
-    val meter: T
+    val timestamp: Instant
 
     val measurements: Collection<Measurement>
+
+    /**
+     * Duplicates the [MeterSnapshot] with a different [meterId] .
```
Comment (oceenachi) [inline api-common/src/main/kotlin/io/qalipsis/api/meters/MeterSnapshot.kt:39]:
```suggestion
     * Duplicates the [MeterSnapshot] with a different [meterId].
```

**Review by oceenachi — CHANGES_REQUESTED** [review]

File: api-common/src/main/kotlin/io/qalipsis/api/meters/MeterType.kt (hunk 2 above)
Comment (ericjesse) [reply api-common/src/main/kotlin/io/qalipsis/api/meters/MeterType.kt:20]: But Possibilities is plural, right?

Comment (oceenachi) [reply api-common/src/main/kotlin/io/qalipsis/api/meters/MeterType.kt:20]: Yea. I think I was looking at "Meter" as singular when I wrote that. But thinking about it again it's fine.

**Review by oceenachi — APPROVED** [review]

**Pull request merged**